flake.nix

```diff
@@ -0,0 +1,164 @@
+        egil-tools-fetch_metadata = import ./nix/egil-tools/fetch_metadata.nix commonArgs;
+        egil-tools-list_metadata = import ./nix/egil-tools/list_metadata.nix commonArgs;
+        egil-tools-public_key_pin = import ./nix/egil-tools/public_key_pin.nix commonArgs;
+      };
+
+      packageNames = attrNames derivations;
+      egilToolPackageNames = filter (hasPrefix "egil-tools-") packageNames;
+      debugPackageNames = filter (hasSuffix "-debug") packageNames;
+      nonDebugPackageNames = subtractLists debugPackageNames packageNames;
+      appPackageNames = subtractLists [ "egil-tools" "egil-plugins-echo" ] packageNames;
+    in
+    {
+      overlays = createOverlays derivations { inherit filterSource; };
+      overlay = self.overlays.egil-scim-client;
+    } // eachSystem supportedSystems (system:
+      let
+        inherit (pkgs.stdenv) isLinux;
+
+        packageNamesToSkip = optionals (system != "x86_64-linux") [ "egil-test-suite" ];
+        packageNamesToKeep = subtractLists packageNamesToSkip packageNames;
+
+        overlays = getAttrs packageNamesToKeep self.overlays;
+        pkgs = import nixpkgs {
+          inherit system;
+          overlays = attrValues overlays;
+        };
+      in
+      rec {
+        checks = packages;
+
+        packages = getAttrs packageNamesToKeep pkgs;
+        defaultPackage = packages.egil-scim-client;
+
+        apps = mapAttrs (name: drv: mkApp { drv = getBin drv; }) (getAttrs (intersectLists packageNamesToKeep appPackageNames) packages);
+        defaultApp = apps.egil-scim-client;
+
+        hydraJobs = {
+          build = getAttrs (intersectLists packageNamesToKeep nonDebugPackageNames) packages;
+
+          vmTest = optionalAttrs isLinux {
+            egil-test-suite = optionalAttrs (system == "x86_64-linux") (import "${nixpkgs}/nixos/tests/make-test-python.nix"
+              ({ ... }: {
+                name = "egil-test-suite";
+
+                machine = { ... }: {
+                  environment.systemPackages = with pkgs; [
+                    egil-test-suite
+                  ];
+
+                  virtualisation.docker.enable = true;
+                  virtualisation.diskSize = 1024; # 512 is not enough
+                };
+
+                testScript = with pkgs; ''
+                  machine.wait_for_unit("docker.service")
+                  machine.execute("${egil-test-suite.loadDockerImages.out}")
+                  machine.succeed("run_test_suite")
+                '';
+              })
+              { inherit system pkgs; });
+          };
+        };
+
+        devShell =
+          let
+            inherit (pkgs) mkShell egil-scim-client-debug;
+            inherit (pkgs.stdenv) glibc;
+
+            debugPackages = attrValues (getAttrs debugPackageNames packages);
+          in
+          mkShell {
+            packages = with pkgs; [
+              gdb.out
+            ] ++ optionals (system == "x86_64-linux") [ egil-test-suite.out ]
```

Comment:
whats up with this?

Reply:
The upstream test suite is only valid for that system because of the docker image dependency.